sklearn/manifold/tests/test_isomap.py

```diff
@@ -20,78 +20,98 @@
 eigen_solvers = ["auto", "dense", "arpack"]
 path_methods = ["auto", "FW", "D"]
 
+N_PER_SIDE = 5
+NPTS = N_PER_SIDE**2
 
-def test_isomap_simple_grid():
-    # Isomap should preserve distances when all neighbors are used
-    N_per_side = 5
-    Npts = N_per_side**2
-    n_neighbors = Npts - 1
 
+def create_sample_data(add_noise=False):
```

Comment:
Can we have `n_per_side` and `n_pts` be parameters for the tests to pass in? This way we can avoid the global parameters.

For example in `test_isomap_simple_grid`, we can set `n_pts=n_neighbors + 1`.